ui/prometheus-plugin/src/plugins/prometheus-time-series-query/get-time-series-data.ts

```diff
@@ -29,6 +29,11 @@ export const getTimeSeriesData: TimeSeriesQueryPlugin<PrometheusTimeSeriesQueryS
   spec,
   context
 ) => {
+  if (spec.query === undefined || spec.query === null || spec.query === '') {
+    // Do not make a request to the backend, instead return an empty TimeSeriesData
+    return { timeRange: { start: new Date(), end: new Date() }, stepMs: 0, series: [] };
```

Comment:
Nice! On a side note, I wonder if we change the plugin implementation to optionally allow these fields? I feel like they don't make sense to be required. cc: @LukeTillman

Reply:
Yeah I'm still playing with this, not sure this is best yet...

Reply:
That's an interesting question. I guess my question would be: if someone were writing a plugin that actually returned series and left those two fields empty/undefined, would the chart still work correctly? If so, totally cool to make them optional. If not, feels like we need some other solution here.

Reply:
@cndonovan we're talking offline but I believe the answer is yes.

`timeRange` should be consumed via the TimeRangeProvider context for better consistency between datasources and panels.
`stepMs` not sure why need this but this can also be computed if not provided.

Reply:
The reason we added it originally was because the time range that really gets used might be different from the system state that was passed in, see:

https://github.com/perses/perses/blob/f0eb9e8ca896b6b158a0f9baac49ddac39f61c10/ui/prometheus-plugin/src/plugins/prometheus-time-series-query/get-time-series-data.ts#L34-L43

Reply:
I see. I think thats fine as metadata but I believe the min-x and max-y in the Chart should still be derived from the Dashboard TimeRange provider